core/src/main/resources/error/error-classes.json

```diff
@@ -290,6 +290,46 @@
           "Null typed values cannot be used as arguments of <functionName>."
         ]
       },
+      "NUMBER_FORMAT_CURRENCY_MUST_BEFORE_DECIMAL" : {
+        "message" : [
+          "Currency characters must appear before any decimal point in the number format: '<numberFormat>'"
```

Comment:
Let's quote the format as we do in other places:
```
toSQLValue(pattern, StringType)
```
see, for instance
https://github.com/apache/spark/blob/0205478b9d35d62450fd7c9ade520087fd2979a7/sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala#L1381

Reply:
Seems there is 5 (or 3) of these, how about using suberror classes here:
INVALID_NUMBER_FORMAT.CURRENCY/EMPTY/SEPARATOR

Or:
INVALID_NUMBER_FORMAT_CURRENT
...

Also I think the two flavors for SEPARATOR and CURRENCY each can be merged.

Currency characters must appear before the decimal point and any digits in the number format

Reply:
Here are all subclasses of DATATYPE_MISMATCH.
Do we really need Level 3 error classes? @MaxGekk 

If we keep Level 2 error classes, INVALID_NUMBER_FORMAT....
The naming rule is inconsistent with other subclasses, eg:
RANGE_FRAME_INVALID_TYPE, RANGE_FRAME_MULTI_ORDER, RANGE_FRAME_WITHOUT_ORDER...

If we merge two flavors for SEPARATOR and CURRENCY, We have to put these characters into the message parameters, which seems inconvenient for subsequent internationalization.

Reply:
> Do we really need Level 3 error classes? @MaxGekk

I don't think it makes sense.

> If we keep Level 2 error classes, INVALID_NUMBER_FORMAT....

Let's make the prefix shorter, for instance `NUM_FORMAT` like
`DATATYPE_MISMATCH.NUM_FORMAT_CONT_THOUSANDS_SEPS`
 instead of
`DATATYPE_MISMATCH.NUMBER_FORMAT_THOUSANDS_SEPARATOR_MUST_HAVE_DIGIT_IN_BETWEEN`